seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/physical/PhysicalVertex.java

```diff
@@ -485,6 +485,8 @@ private void resetExecutionState() {
                         () -> {
                             updateStateTimestamps(ExecutionState.CREATED);
                             runningJobStateIMap.set(taskGroupLocation, ExecutionState.CREATED);
+                            // reset the errorByPhysicalVertex
+                            errorByPhysicalVertex = new AtomicReference<>();
```

Comment:
https://github.com/apache/seatunnel/blob/4b3af9bef4e1753838a7750e86bde71bae8562ae/seatunnel-e2e/seatunnel-e2e-common/src/test/java/org/apache/seatunnel/e2e/sink/inmemory/InMemorySinkWriter.java#L80 `InMemorySinkWriter` is a good choice because it can construct different error messages each time, so you can check the error messages to see if the changes are effective.

Reply:
Ok, I will try it.

Reply:
done